### PR TITLE
hpddm: add new versions

### PR DESCRIPTION
**Pull request by lrtfm**

### CODE REVIEW

**Review by corentin-dev — DISMISSED** [review]
Thank you!

**Comment by corentin-dev** [comment]
Is it working properly for your use case?

**Comment by lrtfm** [comment]
> Is it working properly for your use case?

It can be installed properly. However, it appears that no other packages are dependent on it, is that correct?

I am attempting to establish a dependency of `petsc` on it, but I am unsure of how to do so. https://github.com/spack/spack/discussions/36082#discussioncomment-5581034

**Comment by corentin-dev** [comment]
I added it because I wanted it to be a PETSc dependency first. But I did not achieve it properly because "it is a weird PETSc package" :smiley:

File: var/spack/repos/builtin/packages/hpddm/package.py [hunk 1]
```diff
@@ -16,7 +16,24 @@ class Hpddm(Package):
...
+    version("2.1.0", commit="2096feeca21992be559cfe44de0a196878b19285")
+    version("2.0.8", commit="91e49a61e229462a44acbc0440a9c855b77ff2fd")
+    version("2.0.7", commit="5f2a51beb150eace12bafd884a922c8a0aa332a4")
+    version("2.0.6", commit="ce6ce80b62e9bc71efedf8b9cb6ffb00dfd6e44e")
+    version("2.0.5", commit="21c972f2488edbbd4948c937d7c929b38de9def7")
+    version("2.0.4", commit="90c799d97bd0d3fa738cb288e5111f131b39753b")
+    version("2.0.3", commit="8f545f4217c263fa5c5980b08fa10d68598b5305")
+    version("2.0.2", commit="f94ef3fe56078cc59809f8b8ef568c0e98818e80")
+    version("2.0.1", commit="7d809610de85db8d0c4a53d94ccb188a484ef0f7")
+    version("2.0.0", commit="e7d921ec93b84f41984530822adda1f6fc8b611a")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hpddm/package.py]: Is there a reason to add all the versions and not just the latest?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/hpddm/package.py (hunk 1 above)
Comment (lrtfm) [reply var/spack/repos/builtin/packages/hpddm/package.py]: Packages depending on this will need different versions. Delete the old versions for now.

**Review by alalazo — APPROVED** [review]

**Pull request merged**